include/mbedtls/ssl.h

```diff
@@ -439,16 +439,33 @@
 #define MBEDTLS_SSL_TLS1_3_DEFAULT_NEW_SESSION_TICKETS 1
 #endif
 
-/** \} name SECTION: Module settings */
-
-/*
- * Default to standard CID mode
+/**
```

Comment:
The settings above have no doxygen documentation in `ssl.h`, but have it in `mbedtls_config.h` instead. So I'm not sure I understand the commit message: is it a new rule, or an existing rule that I didn't know and that wasn't consistently followed so far?

Also, why move the documentation specifically for this option but not the others? Overall I'm not sure I understand how this commit relates to the main goal of the PR.

Reply:
It's an existing rule, but not one that we've followed consistently, or ever bothered to write down. Some settings, including all defined-or-not settings, are documented in `mbedtls_config.h` (formerly `config.h`), and `config.py realfull` (formerly `config.pl realfull`) activates all of them so that they appear in the official documentation. Other settings, which are macros with a value, are documented in a topical header file, and wrapped with `#if !defined` in that header file. We don't activate those settings in `realfull` because otherwise Doxygen would see the documentation twice and complain.

Before this PR, documented-in-config.h was based on sections. After this PR, it's mostly based on has-value vs boolean, which I think is easier to understand.

We should make the rule clearer, but that's a preexisting issue outside the scope of this PR, which is about facilitating the implementation of the specification in https://github.com/Mbed-TLS/mbedtls/pull/9236.

As for why we have this rule, I call on the author of 2134d25070a635db3770ec607487380954879499..

Reply:
I'm sorry, I don't get how the rule you mention would follow from https://github.com/Mbed-TLS/mbedtls/commit/2134d25070a635db3770ec607487380954879499 - quite the contrary, the commit message suggests `config.h` had things like:
```
/** Some doxygen doc */
//#define MBEDTLS_FOO
```
which doxygen complains about and we're trying to pacify it by uncommenting the `#define`, not by moving the doxygen comment elsewhere.

I agree that as long as doxygen documentation is generated using `realfull`, macros that are commented out by default in config.h need to have their doxygen doc either in `mbedtls_config.h` or in a topical header (eg `ssl.h`) but not both. I don't see how it logically follows that it must be the 2nd one. I wasn't aware of any rule that it must be the 2nd one, and since you say it's unwritten, I'd say it's hard to know if "we" had such a rule.

Ah, taking a closer look, `realfull` is not doing exactly what I thought it was doing: it's uncommenting everything in most sections and leaving things commented out the "Module configuration options" section, and its docstring says that's because those should be documented in their respective headers - it turns out we have some written rule after all. I'll note that was written by the author of ba4162a52631c0d3ec342a91a0d8924410b4463a - not me. See also the comment maintainers near the start of the "Module configuration options" section, which states something similar but also suggests duplicating the doxygen comment in `config.h`.

Honestly that seems quite complicated. In practice several options don't respect that rule (most SSL options have no doxygen in `ssl.h`) and those that respect the rule have their doxygen duplicated in `mbedtls_config.h` and their topical header, with the the usual issues that come with duplication. I can't see a good reason for that. IMO it would be much simpler for everything to be documented in a single place, and for that single place to be `mbedtls_config.h` - no distinction between sections, experience shows we're not good at remembering those.

Reply:
In ba4162a52631c0d3ec342a91a0d8924410b4463a, I was describing existing behavior, not giving its rationale. I'm now thinking that the current behavior of `realfull` depending on sections is a historical accident and not something we needed for a technical purpose.

Reply:
Yes, I think things evolved in a messy way and a various points one of us tried fixing/improving/documenting things based on the current state at that time, but apparently before today we never took the time to really think about the whole picture :(